lib/iterator.js

```diff
@@ -490,6 +502,39 @@ class SkipWhileIterator extends Iterator {
   }
 }
 
+class KeepLastIterator extends Iterator {
+  constructor(base) {
+    super(base);
+    this.repeat = false;
+    this.lastElement = undefined;
```

Comment:
I don't really like `lastElement` sound like instead of `{ value: T, done: boolean }` it is `T`.